lib/ts/controllers/s-popover.ts

```diff
@@ -278,21 +278,35 @@ export class PopoverController extends BasePopoverController {
     private boundHideOnEscapePress!: any;
 
     /**
-     * Toggles optional classes in addition to BasePopoverController.shown
+     * Toggles optional classes and accessibility attributes in addition to BasePopoverController.shown
      */
-    protected shown(dispatcher: Element|null = null) {
+    protected override shown(dispatcher: Element|null = null) {
         this.toggleOptionalClasses(true);
+        this.toggleAccessibilityAttributes(true);
         super.shown(dispatcher);
     }
 
     /**
-     * Toggles optional classes in addition to BasePopoverController.hidden
+     * Toggles optional classes and accessibility attributes in addition to BasePopoverController.hidden
      */
-    protected hidden(dispatcher: Element|null = null) {
+    protected override hidden(dispatcher: Element|null = null) {
         this.toggleOptionalClasses(false);
+        this.toggleAccessibilityAttributes(false);
         super.hidden(dispatcher);
     }
 
+
+    /**
+     * Initializes accessibility attributes in addition to BasePopoverController.connect
+     */
+    public override connect(): void {
+        super.connect();
+
+        //Make sure the popover is screen reader friendly
+        this.referenceElement.setAttribute("role", "button");
```

Comment:
I'd be curious to get @b-kelly's take, but I might prefer that including this attribute be the responsibility of the consumer of Stacks rather than forcing it.

I can't think of an edge case that this will cause issues with but could their be a case where adding the role is unwanted and the consumer would have to write additional JS to remove it?

Reply:
I definitely understand the concern. 

From the literature I've read there are a few conditions that need to be met for a popover to be considered accessible. Not sure how many of them we can 'force' like this on the Stacks side: https://www.w3.org/TR/wai-aria-practices-1.1/#wai-aria-roles-states-and-properties-14

For example it also states that we need to add the `aria-haspopup` attribute but I also shared some concerns around that [below](https://github.com/StackExchange/Stacks/pull/932#issuecomment-1119979072). So maybe these parts of the markup can be left to the author like you said. Happy to go either way.

Reply:
Thanks for the feedback Ben/Dan. I went ahead and removed the role assignment.

Just some more thoughts I wanted to share here:
From what I've read if the popover triggering element is clickable it should have the `role="button"` attribute regardless of which type of html element it is. But in cases where it's not clickable and only triggerable elsewhere via JS then I agree the role shouldn't be there although I've also read that that isn't very good design in terms of accessibility (Deque's documentation mentions how UI's need to be "predictable"). But we can't predict all use cases like you said.

Reply:
I think updating the documentation would be better served as a follow up PR. I won't be able to give it the care it needs right now and don't want to push for a half-baked solution.

Reply:
Just had a random thought as well. Have we considered adding a "Accessibility Considerations" section to our Stacks components pages? We could have one on the [Popover component page](https://stackoverflow.design/product/components/popovers/) for example.

We could lay out details on accessibility there to inform users on the how and why. The reason I think that may be important is because there's no cookie cutter solution for everything so each UI element needs to be considered in its own context and the solution may vary accordingly. 

Although having a section like this would be more maintenance work for us as the standards around accessibility change over time.

Reply:
I would absolutely consider having an "Accessibility" section on the component pages. We already have one on the [modals page](https://stackoverflow.design/product/components/modals/#accessibility) already, so there's already some prior art there. That's essentially what I was envisioning with the popovers section.

Reply:
Ooh I hadn't noticed that. Probably because I never had to add a modal component, lol. That's awesome. I'll try and look into adding a section like that for popovers when I can 👍🏼

Reply:
No prob if you don't get to it - it isn't required for this PR to be merged.